src/python/pants/backend/python/macros/python_artifact.py

```diff
@@ -77,7 +78,12 @@ def __init__(self, **kwargs):
             # coerce entry points from Dict[str, List[str]] to Dict[str, Dict[str, str]]
             kwargs["entry_points"] = _normalize_entry_points(kwargs["entry_points"])
 
-        self._kw: Dict[str, Any] = kwargs
+        self._kw: Dict[str, Any] = copy.deepcopy(kwargs)
+        # The kwargs come from a BUILD file, and can contain somewhat arbitrary nested structures,
+        # so we don't have a principled way to make them into a hashable data structure.
+        # Instead we stringify and precompute a hash to use in our own __hash__, since we know
+        # that our
+        self._hash: int = hash(str(kwargs))
         self._binaries = {}
```

Comment:
This is worrying. It seems like the `binaries` method below shoud be returning a copy or else the equality / hash non-importance of that externally mutable field should probably be comment addressed.

Reply:
Turns out that is unused and can be removed entirely!